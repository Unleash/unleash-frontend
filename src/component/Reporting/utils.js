import parseISO from 'date-fns/parseISO';
import differenceInDays from 'date-fns/differenceInDays';

import { EXPERIMENT, OPERATIONAL, RELEASE, FOURTYDAYS, SEVENDAYS } from './constants';

export const toggleExpiryByTypeMap = {
    [EXPERIMENT]: FOURTYDAYS,
    [RELEASE]: FOURTYDAYS,
    [OPERATIONAL]: SEVENDAYS,
};

export const applyCheckedToFeatures = (features, checkedState) =>
    features.map(feature => ({ ...feature, checked: checkedState }));

export const getCheckedState = (name, features) => {
    const feature = features.find(feature => feature.name === name);

    if (feature) {
        return feature.checked ? feature.checked : false;
    }
    return false;
};

export const getDiffInDays = (date, now) => Math.abs(differenceInDays(date, now));

export const formatProjectOptions = projects => projects.map(project => ({ key: project.id, label: project.name }));

export const expired = (diff, type) => {
    if (diff >= toggleExpiryByTypeMap[type]) return true;
    return false;
};

export const getObjectProperties = (target, ...keys) => {
    const newObject = {};

    keys.forEach(key => {
        if (target[key] !== undefined) {
            newObject[key] = target[key];
        }
    });

    return newObject;
};

export const sortFeaturesByNameAscending = features => {
    const sorted = [...features];
    sorted.sort((a, b) => {
        if (a.name < b.name) {
            return -1;
        }
        if (a.name > b.name) {
            return 1;
        }
        return 0;
    });
    return sorted;
};

export const sortFeaturesByNameDescending = features => sortFeaturesByNameAscending([...features]).reverse();

export const sortFeaturesByLastSeenAscending = features => {
    const sorted = [...features];
    sorted.sort((a, b) => {
        if (!a.lastSeenAt) return -1;
        if (!b.lastSeenAt) return 1;

        const dateA = parseISO(a.lastSeenAt);
        const dateB = parseISO(b.lastSeenAt);

        return dateA.getTime() - dateB.getTime();
    });
    return sorted;
};

export const sortFeaturesByLastSeenDescending = features => sortFeaturesByLastSeenAscending([...features]).reverse();

export const sortFeaturesByCreatedAtAscending = features => {
    const sorted = [...features];
    sorted.sort((a, b) => {
        const dateA = parseISO(a.createdAt);
        const dateB = parseISO(b.createdAt);

        return dateA.getTime() - dateB.getTime();
    });
    return sorted;
};

export const sortFeaturesByCreatedAtDescending = features => sortFeaturesByCreatedAtAscending([...features]).reverse();

export const sortFeaturesByExpiredAtAscending = features => {
    const sorted = [...features];
    sorted.sort((a, b) => {
        const now = new Date();
        const dateA = parseISO(a.createdAt);
        const dateB = parseISO(b.createdAt);

        const diffA = getDiffInDays(dateA, now);
        const diffB = getDiffInDays(dateB, now);

        if (!expired(diffA, a.type)) return -1;
        if (!expired(diffB, b.type)) return 1;

        console.log('EXPIRED');

        const expiredByA = diffA - toggleExpiryByTypeMap[a.type];
        const expiredByB = diffB - toggleExpiryByTypeMap[b.type];

        return expiredByA - expiredByB;
    });
    return sorted;
};

export const sortFeaturesByExpiredAtDescending = features => sortFeaturesByExpiredAtAscending([...features]).reverse();

export const sortFeaturesByStatusAscending = features => {
    const sorted = [...features];
    sorted.sort((a, b) => {
        if (a.stale) return 1;
        if (b.stale) return -1;
        return 0;
    });
    return sorted;
};

export const sortFeaturesByStatusDescending = features => sortFeaturesByStatusAscending([...features]).reverse();

export const pluralize = (items, word) => {
    if (items === 1) return `${items} ${word}`;
    return `${items} ${word}s`;
};

export const getDates = dateString => {
    const date = parseISO(dateString);
    const now = new Date();

    return [date, now];
};

export const filterByProject = selectedProject => feature => feature.project === selectedProject;

export const isFeatureExpired = feature => {
    const [date, now] = getDates(feature.createdAt);
    const diff = getDiffInDays(date, now);

    return expired(diff, feature.type);
};
