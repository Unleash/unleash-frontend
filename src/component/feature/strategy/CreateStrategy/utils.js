export const resolveDefaultParamValue = (name, featureToggleName) => {
    switch (name) {
        case "percentage":
        case "rollout":
            return "100";
        case "stickiness":
            return "default";
        case "groupId":
            return featureToggleName;
        default:
            return "";
    }
};
