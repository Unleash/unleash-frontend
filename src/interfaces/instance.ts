export interface InstanceStatus {
    plan: string;
    trialExpiry?: string;
    trialStart?: string;
    trialExtended?: number;
    billingCenter?: string;
    instanceState?: InstanceState;
}

export enum InstanceState {
    UNASSIGNED = 'UNASSIGNED',
    TRIAL = 'TRIAL',
    ACTIVE = 'ACTIVE',
    EXPIRED = 'EXPIRED',
    CHURNED = 'CHURNED',
}
