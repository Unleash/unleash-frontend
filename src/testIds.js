/* NAVIGATION */
export const NAVIGATE_TO_CREATE_FEATURE = 'NAVIGATE_TO_CREATE_FEATURE';
export const CREATE_API_TOKEN_BUTTON = 'CREATE_API_TOKEN_BUTTON';

/* CREATE FEATURE */
export const CF_NAME_ID = 'CF_NAME_ID';
export const CF_TYPE_ID = 'CF_TYPE_ID';
export const CF_DESC_ID = 'CF_DESC_ID';
export const CF_CREATE_BTN_ID = 'CF_CREATE_BTN_ID';

/* LOGIN */
export const LOGIN_EMAIL_ID = 'LOGIN_EMAIL_ID';
export const LOGIN_BUTTON = 'LOGIN_BUTTON';
export const LOGIN_PASSWORD_ID = 'LOGIN_PASSWORD_ID';
export const SSO_LOGIN_BUTTON = 'SSO_LOGIN_BUTTON';
export const FORGOTTEN_PASSWORD_FIELD = 'FORGOTTEN_PASSWORD_FIELD';

/* STRATEGY */
export const ADD_NEW_STRATEGY_ID = 'ADD_NEW_STRATEGY_ID';
export const ADD_NEW_STRATEGY_CARD_BUTTON_ID =
    'ADD_NEW_STRATEGY_CARD_BUTTON_ID';
export const ROLLOUT_SLIDER_ID = 'ROLLOUT_SLIDER_ID';
export const ADD_NEW_STRATEGY_SAVE_ID = 'ADD_NEW_STRATEGY_SAVE_ID';
export const DIALOGUE_CONFIRM_ID = 'DIALOGUE_CONFIRM_ID';
export const ADD_CONSTRAINT_ID = 'ADD_CONSTRAINT_ID';
export const CONSTRAINT_AUTOCOMPLETE_ID = 'CONSTRAINT_AUTOCOMPLETE_ID';
export const STRATEGY_ACCORDION_ID = 'STRATEGY_ACCORDION_ID';
export const FLEXIBLE_STRATEGY_STICKINESS_ID =
    'FLEXIBLE_STRATEGY_STICKINESS_ID';
export const FLEXIBLE_STRATEGY_GROUP_ID = 'FLEXIBLE_STRATEGY_GROUP_ID';
export const SELECT_ITEM_ID = 'SELECT_ITEM_ID';
export const UPDATE_STRATEGY_BUTTON_ID = 'UPDATE_STRATEGY_BUTTON_ID';
export const DELETE_STRATEGY_ID = 'DELETE_STRATEGY_ID';
export const STRATEGY_INPUT_LIST = 'STRATEGY_INPUT_LIST';
export const ADD_TO_STRATEGY_INPUT_LIST = 'ADD_TO_STRATEGY_INPUT_LIST';
export const STRATEGY_FORM_SUBMIT_ID = 'STRATEGY_FORM_SUBMIT_ID';
export const STRATEGY_FORM_REMOVE_ID = 'STRATEGY_FORM_REMOVE_ID';

/* SPLASH */
export const CLOSE_SPLASH = 'CLOSE_SPLASH';

/* GENERAL */
export const INPUT_ERROR_TEXT = 'INPUT_ERROR_TEXT';
export const HEADER_USER_AVATAR = 'HEADER_USER_AVATAR';
