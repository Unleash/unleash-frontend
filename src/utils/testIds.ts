/* NAVIGATION */
export const NAVIGATE_TO_CREATE_FEATURE = 'NAVIGATE_TO_CREATE_FEATURE';
export const NAVIGATE_TO_CREATE_SEGMENT = 'NAVIGATE_TO_CREATE_SEGMENT';
export const CREATE_API_TOKEN_BUTTON = 'CREATE_API_TOKEN_BUTTON';

/* CREATE FEATURE */
export const CF_NAME_ID = 'CF_NAME_ID';
export const CF_TYPE_ID = 'CF_TYPE_ID';
export const CF_DESC_ID = 'CF_DESC_ID';
export const CF_CREATE_BTN_ID = 'CF_CREATE_BTN_ID';

/* SEGMENT */
export const SEGMENT_NAME_ID = 'SEGMENT_NAME_ID';
export const SEGMENT_DESC_ID = 'SEGMENT_DESC_ID';
export const SEGMENT_NEXT_BTN_ID = 'SEGMENT_NEXT_BTN_ID';
export const SEGMENT_CREATE_BTN_ID = 'SEGMENT_CREATE_BTN_ID';
export const SEGMENT_SAVE_BTN_ID = 'SEGMENT_SAVE_BTN_ID';
export const SEGMENT_DELETE_BTN_ID = 'SEGMENT_DELETE_BTN_ID';
export const SEGMENT_DIALOG_NAME_ID = 'SEGMENT_DIALOG_NAME_ID';

/* LOGIN */
export const LOGIN_EMAIL_ID = 'LOGIN_EMAIL_ID';
export const LOGIN_BUTTON = 'LOGIN_BUTTON';
export const LOGIN_PASSWORD_ID = 'LOGIN_PASSWORD_ID';
export const SSO_LOGIN_BUTTON = 'SSO_LOGIN_BUTTON';
export const FORGOTTEN_PASSWORD_FIELD = 'FORGOTTEN_PASSWORD_FIELD';
export const INVALID_TOKEN_BUTTON = 'INVALID_TOKEN_BUTTON';

/* STRATEGY */
export const FEATURE_ENVIRONMENT_ACCORDION = 'FEATURE_ENVIRONMENT_ACCORDION';
export const ADD_NEW_STRATEGY_ID = 'ADD_NEW_STRATEGY_ID';
export const ROLLOUT_SLIDER_ID = 'ROLLOUT_SLIDER_ID';
export const DIALOGUE_CONFIRM_ID = 'DIALOGUE_CONFIRM_ID';
export const CONSTRAINT_AUTOCOMPLETE_ID = 'CONSTRAINT_AUTOCOMPLETE_ID';
export const FLEXIBLE_STRATEGY_STICKINESS_ID =
    'FLEXIBLE_STRATEGY_STICKINESS_ID';
export const FLEXIBLE_STRATEGY_GROUP_ID = 'FLEXIBLE_STRATEGY_GROUP_ID';
export const SELECT_ITEM_ID = 'SELECT_ITEM_ID';
export const STRATEGY_INPUT_LIST = 'STRATEGY_INPUT_LIST';
export const ADD_TO_STRATEGY_INPUT_LIST = 'ADD_TO_STRATEGY_INPUT_LIST';
export const STRATEGY_FORM_SUBMIT_ID = 'STRATEGY_FORM_SUBMIT_ID';
export const STRATEGY_FORM_REMOVE_ID = 'STRATEGY_FORM_REMOVE_ID';

/* SPLASH */
export const CLOSE_SPLASH = 'CLOSE_SPLASH';

/* GENERAL */
export const INPUT_ERROR_TEXT = 'INPUT_ERROR_TEXT';
export const HEADER_USER_AVATAR = 'HEADER_USER_AVATAR';
export const SIDEBAR_MODAL_ID = 'SIDEBAR_MODAL_ID';
export const AUTH_PAGE_ID = 'AUTH_PAGE_ID';
