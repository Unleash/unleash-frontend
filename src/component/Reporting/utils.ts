import parseISO from 'date-fns/parseISO';
import differenceInDays from 'date-fns/differenceInDays';

import { EXPERIMENT, OPERATIONAL, RELEASE } from 'constants/featureToggleTypes';

import { FOURTYDAYS, SEVENDAYS } from './constants';
import { IFeatureToggleListItem } from 'interfaces/featureToggle';

export const toggleExpiryByTypeMap: Record<string, number> = {
    [EXPERIMENT]: FOURTYDAYS,
    [RELEASE]: FOURTYDAYS,
    [OPERATIONAL]: SEVENDAYS,
};

export interface IFeatureToggleListItemCheck extends IFeatureToggleListItem {
    checked: boolean;
}

export const applyCheckedToFeatures = (
    features: IFeatureToggleListItem[],
    checkedState: boolean
): IFeatureToggleListItemCheck[] => {
    return features.map(feature => ({
        ...feature,
        checked: checkedState,
    }));
};

export const getCheckedState = (
    name: string,
    features: IFeatureToggleListItemCheck[]
) => {
    const feature = features.find(feature => feature.name === name);

    if (feature) {
        return feature.checked;
    }

    return false;
};

export const getDiffInDays = (date: Date, now: Date) =>
    Math.abs(differenceInDays(date, now));

export const expired = (diff: number, type: string) => {
    if (diff >= toggleExpiryByTypeMap[type]) return true;
    return false;
};

export const getObjectProperties = <T extends object>(
    target: T,
    ...keys: (keyof T)[]
): Partial<T> => {
    const newObject: Partial<T> = {};

    keys.forEach(key => {
        if (target[key] !== undefined) {
            newObject[key] = target[key];
        }
    });

    return newObject;
};

export const sortFeaturesByNameAscending = (
    features: IFeatureToggleListItem[]
): IFeatureToggleListItem[] => {
    const sorted = [...features];
    sorted.sort((a, b) => {
        if (a.name < b.name) {
            return -1;
        }
        if (a.name > b.name) {
            return 1;
        }
        return 0;
    });
    return sorted;
};

export const sortFeaturesByNameDescending = (
    features: IFeatureToggleListItem[]
): IFeatureToggleListItem[] =>
    sortFeaturesByNameAscending([...features]).reverse();

export const sortFeaturesByLastSeenAscending = (
    features: IFeatureToggleListItem[]
): IFeatureToggleListItem[] => {
    const sorted = [...features];
    sorted.sort((a, b) => {
        if (!a.lastSeenAt) return -1;
        if (!b.lastSeenAt) return 1;

        const dateA = parseISO(a.lastSeenAt);
        const dateB = parseISO(b.lastSeenAt);

        return dateA.getTime() - dateB.getTime();
    });
    return sorted;
};

export const sortFeaturesByLastSeenDescending = (
    features: IFeatureToggleListItem[]
): IFeatureToggleListItem[] =>
    sortFeaturesByLastSeenAscending([...features]).reverse();

export const sortFeaturesByCreatedAtAscending = (
    features: IFeatureToggleListItem[]
): IFeatureToggleListItem[] => {
    const sorted = [...features];
    sorted.sort((a, b) => {
        const dateA = parseISO(a.createdAt);
        const dateB = parseISO(b.createdAt);

        return dateA.getTime() - dateB.getTime();
    });
    return sorted;
};

export const sortFeaturesByCreatedAtDescending = (
    features: IFeatureToggleListItem[]
): IFeatureToggleListItem[] =>
    sortFeaturesByCreatedAtAscending([...features]).reverse();

export const sortFeaturesByExpiredAtAscending = (
    features: IFeatureToggleListItem[]
): IFeatureToggleListItem[] => {
    const sorted = [...features];
    sorted.sort((a, b) => {
        const now = new Date();
        const dateA = parseISO(a.createdAt);
        const dateB = parseISO(b.createdAt);

        const diffA = getDiffInDays(dateA, now);
        const diffB = getDiffInDays(dateB, now);

        if (!expired(diffA, a.type) && expired(diffB, b.type)) {
            return 1;
        }

        if (expired(diffA, a.type) && !expired(diffB, b.type)) {
            return -1;
        }

        const expiredByA = diffA - toggleExpiryByTypeMap[a.type];
        const expiredByB = diffB - toggleExpiryByTypeMap[b.type];

        return expiredByB - expiredByA;
    });
    return sorted;
};

export const sortFeaturesByExpiredAtDescending = (
    features: IFeatureToggleListItem[]
): IFeatureToggleListItem[] => {
    const sorted = [...features];
    const now = new Date();
    sorted.sort((a, b) => {
        const dateA = parseISO(a.createdAt);
        const dateB = parseISO(b.createdAt);

        const diffA = getDiffInDays(dateA, now);
        const diffB = getDiffInDays(dateB, now);

        if (!expired(diffA, a.type) && expired(diffB, b.type)) {
            return 1;
        }

        if (expired(diffA, a.type) && !expired(diffB, b.type)) {
            return -1;
        }

        const expiredByA = diffA - toggleExpiryByTypeMap[a.type];
        const expiredByB = diffB - toggleExpiryByTypeMap[b.type];

        return expiredByA - expiredByB;
    });
    return sorted;
};

export const sortFeaturesByStatusAscending = (
    features: IFeatureToggleListItem[]
): IFeatureToggleListItem[] => {
    const sorted = [...features];
    sorted.sort((a, b) => {
        if (a.stale) return 1;
        if (b.stale) return -1;
        return 0;
    });
    return sorted;
};

export const sortFeaturesByStatusDescending = (
    features: IFeatureToggleListItem[]
): IFeatureToggleListItem[] =>
    sortFeaturesByStatusAscending([...features]).reverse();

export const pluralize = (items: number, word: string): string => {
    if (items === 1) return `${items} ${word}`;
    return `${items} ${word}s`;
};

export const getDates = (dateString: string): [Date, Date] => {
    const date = parseISO(dateString);
    const now = new Date();

    return [date, now];
};
