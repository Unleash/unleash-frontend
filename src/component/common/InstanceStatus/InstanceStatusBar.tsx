import { styled, Button } from '@mui/material';
import { colors } from 'themes/colors';
import { IInstanceStatus, InstanceState } from 'interfaces/instance';
import { differenceInDays, parseISO } from 'date-fns';
import { INSTANCE_STATUS_BAR_ID } from 'utils/testIds';
import { Info } from '@mui/icons-material';

interface IInstanceStatusBarProps {
    instanceStatus: IInstanceStatus;
}

export const InstanceStatusBar = ({
    instanceStatus,
}: IInstanceStatusBarProps) => {
    const trialDaysRemaining = calculateTrialDaysRemaining(instanceStatus);

    if (
        instanceStatus.instanceState === InstanceState.TRIAL &&
        typeof trialDaysRemaining === 'number' &&
        trialDaysRemaining <= 0
    ) {
        return (
            <StyledBar data-testid={INSTANCE_STATUS_BAR_ID}>
                <StyledInfoIcon />
                <span>
                    <strong>Heads up!</strong> Your free trial of the{' '}
                    {instanceStatus.plan.toUpperCase()} version has expired.
                </span>
                <ContactButton />
            </StyledBar>
        );
    }

    if (
        instanceStatus.instanceState === InstanceState.TRIAL &&
        typeof trialDaysRemaining === 'number' &&
        trialDaysRemaining <= 10
    ) {
        return (
            <StyledBar data-testid={INSTANCE_STATUS_BAR_ID}>
                <StyledInfoIcon />
                <span>
                    <strong>Heads up!</strong> You have{' '}
                    <strong>{trialDaysRemaining} days</strong> remaining of your
                    free trial of the {instanceStatus.plan.toUpperCase()}{' '}
                    version.
                </span>
                <ContactButton />
            </StyledBar>
        );
    }

    return null;
};

const ContactButton = () => {
    return (
        <StyledButton
            href="mailto:support@getunleash.zendesk.com"
            variant="outlined"
        >
            Contact us
        </StyledButton>
    );
};

const calculateTrialDaysRemaining = (
    instanceStatus: IInstanceStatus
): number | undefined => {
    return instanceStatus.trialExpiry
        ? differenceInDays(parseISO(instanceStatus.trialExpiry), new Date())
        : undefined;
};

const StyledBar = styled('aside')(({ theme }) => ({
    position: 'relative',
    zIndex: 1,
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center',
    padding: theme.spacing(2),
    gap: theme.spacing(1),
    borderBottom: '1px solid',
    borderColor: colors.blue[200],
    background: colors.blue[50],
    color: colors.blue[700],
}));

const StyledButton = styled(Button)(({ theme }) => ({
    whiteSpace: 'nowrap',
    minWidth: '8rem',
    marginLeft: theme.spacing(2),
}));

const StyledInfoIcon = styled(Info)(({ theme }) => ({
    color: colors.blue[500],
}));
