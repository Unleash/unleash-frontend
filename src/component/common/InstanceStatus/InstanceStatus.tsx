import { useInstanceStatus } from 'hooks/api/getters/useInstanceStatus/useInstanceStatus';
import React, { ReactNode } from 'react';
import { InstanceStatusBar } from 'component/common/InstanceStatus/InstanceStatusBar';

interface IInstanceStatusProps {
    children: ReactNode;
}

export const InstanceStatus = ({ children }: IInstanceStatusProps) => {
    const { instanceStatus } = useInstanceStatus();

    return (
        <div hidden={!instanceStatus} style={{ height: '100%' }}>
            {instanceStatus && (
                <InstanceStatusBarMemo instanceStatus={instanceStatus} />
            )}
            {children}
        </div>
    );
};

const InstanceStatusBarMemo = React.memo(InstanceStatusBar);
