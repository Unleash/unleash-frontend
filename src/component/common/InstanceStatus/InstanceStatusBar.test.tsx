import { InstanceStatusBar } from 'component/common/InstanceStatus/InstanceStatusBar';
import { InstanceState } from 'interfaces/instance';
import { render } from 'utils/testRenderer';
import { screen } from '@testing-library/react';
import { addDays } from 'date-fns';
import { INSTANCE_STATUS_BAR_ID } from 'utils/testIds';

test('InstanceStatusBar should be hidden by default', async () => {
    render(<InstanceStatusBar instanceStatus={{ plan: 'pro' }} />);

    expect(
        screen.queryByTestId(INSTANCE_STATUS_BAR_ID)
    ).not.toBeInTheDocument();
});

test('InstanceStatusBar should warn when the trial is about to expire', async () => {
    render(
        <InstanceStatusBar
            instanceStatus={{
                plan: 'pro',
                instanceState: InstanceState.TRIAL,
                trialExpiry: addDays(new Date(), 5).toISOString(),
            }}
        />
    );

    expect(screen.getByTestId(INSTANCE_STATUS_BAR_ID)).toBeInTheDocument();
    expect(await screen.findByTestId(INSTANCE_STATUS_BAR_ID)).toMatchSnapshot();
});

test('InstanceStatusBar should be hidden when the trial has expired', async () => {
    render(
        <InstanceStatusBar
            instanceStatus={{
                plan: 'pro',
                instanceState: InstanceState.TRIAL,
                trialExpiry: new Date().toISOString(),
            }}
        />
    );

    expect(
        screen.queryByTestId(INSTANCE_STATUS_BAR_ID)
    ).not.toBeInTheDocument();
});
