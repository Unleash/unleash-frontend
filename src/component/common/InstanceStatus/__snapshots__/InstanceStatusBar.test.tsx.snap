// Vitest Snapshot v1

exports[`InstanceStatusBar should warn when the trial is about to expire 1`] = `
<aside
  class="mui-1rw10cs"
  data-testid="INSTANCE_STATUS_BAR_ID"
>
  <svg
    aria-hidden="true"
    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium mui-cle2im-MuiSvgIcon-root"
    data-testid="InfoIcon"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <path
      d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 15h-2v-6h2v6zm0-8h-2V7h2v2z"
    />
  </svg>
  <span>
    <strong>
      Heads up!
    </strong>
     You have
     
    <strong>
      4
       days
    </strong>
     remaining of your free trial of the 
    PRO
     
    version.
  </span>
  <a
    class="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonBase-root mui-l66s6r-MuiButtonBase-root-MuiButton-root"
    href="mailto:support@getunleash.zendesk.com"
    tabindex="0"
  >
    Contact us
    <span
      class="MuiTouchRipple-root mui-8je8zh-MuiTouchRipple-root"
    />
  </a>
</aside>
`;
